### PR TITLE
(fix) Passing context

### PR DESCRIPTION
**Pull request by stanislavlysenko0912**
Issue #348 

Added the feature for the bot to see the context when used:
 - /chat
 - /chat "GROUP_TRIGGER_KEYWORD"

image

### CODE REVIEW

**Comment by stanislavlysenko0912** [comment]
Perhaps an implementation with a modification of the "message_text" function would be more universal, which would unpack the command and the Prompt, but for the sake of such a function I think there is no point in modifying it

File: bot/telegram_bot.py [hunk 1]
```diff
@@ -370,7 +370,14 @@ async def prompt(self, update: Update, context: ContextTypes.DEFAULT_TYPE):
...
             trigger_keyword = self.config['group_trigger_keyword']
-            if prompt.lower().startswith(trigger_keyword.lower()):
+
+            raw_text = update.message.text.lower()
+            prompt_lower = prompt.lower()
+
+            if raw_text.startswith('/chat') or prompt_lower.startswith(trigger_keyword):
+                if not prompt_lower.startswith(trigger_keyword):
+                    prompt = f'{trigger_keyword} {prompt}'
+
```
Comment (n3d1117) [inline bot/telegram_bot.py:380]: Could this work the same way?

```suggestion
            if prompt.lower().startswith(trigger_keyword.lower()) or update.message.text.lower().startswith('/chat'):
                if prompt.lower().startswith(trigger_keyword.lower()):
                    prompt = prompt[len(trigger_keyword):].strip()
```

**Comment by n3d1117** [comment]
Hey @stanislavlysenko0912, apologies for the super late reply! Thanks for opening this, just checked and I think the fix could be made even simpler. I've commented the suggestion, let me know!

**Comment by stanislavlysenko0912** [comment]
> Hey @stanislavlysenko0912, apologies for the super late reply! Thanks for opening this, just checked and I think the fix could be made even simpler. I've commented the suggestion, let me know!

I'm sorry I answered the wrong place, just in the thread)
Anyway, I think you'll see what I mean. 

I could take the time to look for another way to implement, if you want - but so far nothing comes to mind :D

**Comment by n3d1117** [comment]
> I'm sorry I answered the wrong place, just in the thread) Anyway, I think you'll see what I mean.

Huh? Sorry I don't get it, did you answer somewhere else?

> I could take the time to look for another way to implement, if you want - but so far nothing comes to mind :D

Nothing wrong with your original PR, but could you check out the fix I suggested in the review above?

It's basically this:
```python
if prompt.lower().startswith(trigger_keyword.lower()) or update.message.text.lower().startswith('/chat'):
      if prompt.lower().startswith(trigger_keyword.lower()):
          prompt = prompt[len(trigger_keyword):].strip()
```

instead of:

```python
raw_text = update.message.text.lower()
prompt_lower = prompt.lower()
if raw_text.startswith('/chat') or prompt_lower.startswith(trigger_keyword):
    if not prompt_lower.startswith(trigger_keyword):
        prompt = f'{trigger_keyword} {prompt}'
```

**Comment by stanislavlysenko0912** [comment]
@n3d1117 

Your code will be work correctly, if remove this line after condition

```python
trigger_keyword = self.config['group_trigger_keyword']

if prompt.lower().startswith(trigger_keyword.lower()) or update.message.text.lower().startswith('/chat'):
    if prompt.lower().startswith(trigger_keyword.lower()):
        prompt = prompt[len(trigger_keyword):].strip()

    prompt = prompt[len(trigger_keyword):].strip() # delete this line, because it will be remove first symbol every time

    if update.message.reply_to_message and \
            update.message.reply_to_message.text and \
            update.message.reply_to_message.from_user.id != context.bot.id:
        prompt = f'"{update.message.reply_to_message.text}" {prompt}'
```

**Comment by n3d1117** [comment]
Thanks @stanislavlysenko0912!

**Pull request merged**